lib/scripts/read_procedures_pdf.rb

```diff
@@ -0,0 +1,87 @@
+# frozen_string_literal: true
+
+require "csv"
+
+module Scripts
+  class ReadProceduresPDF
+    attr_reader :path_file, :batch_size
+
+    def initialize(path_file, batch_size)
+      @path_file = path_file
+      @batch_size = batch_size
+    end
+
+    def run
+      csv_data_to_json!
+    end
+
+    private
+
+    def csv_data_to_json!
+      file.each_slice(batch_size).with_index do |batch, batch_index|
+        procedures_hash = { batch: { procedures: [] } }
+
+        batch.each_with_index do |row, index|
+          args = { procedures_hash: procedures_hash, row: row, index: index, batch_index: batch_index }
+
+          begin
+            validate_data_row!(row)
+            from_row_to_json(args)
+          rescue StandardError => e
+            raise StandardError, e.message
+          end
+        end
+
+        batches_procedures_parsed(batch_index, procedures_hash.to_json)
+      end
+    end
+
+    def validate_data_row!(row)
+      return if row.blank?
+
+      validate_code!(row[0])
+      validate_port!(row[4])
+    end
+
+    def validate_code!(code)
+      return if code.blank?
+      return true if /^\d+$/.match?(code)
```

Comment:
I think you can name this regex

Reply:
Done! [Commit](https://github.com/espoo-dev/med_system_backend/pull/165/commits/ded9701830f0ee6be3d55da258161b3a806b8bd5) `ded9701830f0ee6be3d55da258161b3a806b8bd5`